src/application/ApplicationDetail/panels/ActivityLogPanel.tsx

```diff
@@ -0,0 +1,220 @@
+import moment from 'moment';
+import React, { Component } from 'react';
+import { PushApplication } from '@aerogear/unifiedpush-admin-client';
+import {
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Text,
+  Label,
+  Alert,
+  Pagination,
+  PaginationVariant,
+} from '@patternfly/react-core';
+import { TableIcon, ErrorCircleOIcon } from '@patternfly/react-icons';
+import { UpsClientFactory } from '../../../utils/UpsClientFactory';
+import { FlatPushMessageInformation } from '@aerogear/unifiedpush-admin-client/dist/src/commands/metrics/LoadMetricsCommand';
+import { Title } from '../../../common/Title';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  ICell,
+  IRow,
+  expandable,
+} from '@patternfly/react-table';
+import { EllipsisText } from '../../../common/EllipsisText';
+import { CodeSnippet } from '../CodeSnippet';
+
+interface Props {
+  app: PushApplication;
+}
+
+interface State {
+  loading: boolean;
+  log: FlatPushMessageInformation[];
+  columns: ICell[];
+  rows: IRow[];
+  total: number;
+  currentPage: number;
+  perPage: number;
+}
+
+export class ActivityLogPanel extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      currentPage: 1,
+      perPage: 10,
+      loading: true,
+      total: 0,
+      log: [],
+      columns: [
+        { title: 'Message', cellFormatters: [expandable] },
+        { title: 'IP Address' },
+        { title: 'Status' },
+        { title: 'Sent' },
+      ],
+      rows: [],
+    };
+
+    this.onCollapse = this.onCollapse.bind(this);
+  }
+
+  loadActivityLog = async (page = 0) => {
+    const makeDetail = (
+      message: string,
+      metric: FlatPushMessageInformation
+    ): React.ReactNode => {
+      let errorElement = <></>;
+
+      if (metric.errors && metric.errors[0]) {
+        errorElement = (
+          <Alert
+            variant="warning"
```

Comment:
I believe this should be an inline alert (add `isInline`: https://www.patternfly.org/v4/documentation/react/components/alert#inline-types)